第6課題　SpringBootを使ってHelloWorldの表示＋時刻表示

toritoripi opened the pull request:
時刻表示の機能を追加しています。
よろしくお願いします。

toritoripi committed:
Add files via upload

toritoripi committed:
時刻表示の追加

toritoripi committed:
Create README.md

toritoripi committed:
Update README.md

toritoripi committed:
Update README.md

toritoripi committed:
Update README.md

toritoripi committed:
Update README.md

yoshi-koyama commented on HelloWorldController.java:
```diff
@@ -3,6 +3,11 @@
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.time.format.DateTimeFormatter;
```
Pull Requestの概要に動作確認の内容を貼ってもらえるとありがたいです！

yoshi-koyama commented on HelloWorldController.java:
```diff
@@ -12,4 +17,15 @@ public String hello() {
+    @GetMapping("/time")
+    public Map<String, String> time() {
+        Map<String, String> replyMap = new HashMap<String, String>();
+        replyMap.put("Greeting", "hello world!");
```
jsonの命名規則について調べていただきたいです。
おそらくsnake_caseかcamelCaseが多いはず。
となるとGreetingも修正が必要になるはずです。

toritoripi replied:
```suggestion
        replyMap.put("greeting", "hello world!");

```
snake_caseにもとづいて小文字表記にしました。

yoshi-koyama replied:
治ったと思いましたが、push漏れかな？

yoshi-koyama replied:
あ、ちがいました。このHelloWorldController.javaは削除しましょう。

yoshi-koyama commented on README.md:
```diff
@@ -0,0 +1,8 @@
+# 第6課題
+SpringBootを使ってHelloWorldの表示、時刻表示
+# 概要
+？？
```
これはあえて`??`にしていますか？

toritoripi replied:
概要欄に記入することがまとめられず（細かい部分で自分がなにをやっているのか言語化できない）？にしてしまいました。後日にでも整理してきちんと記入したいと思っています。

yoshi-koyama commented on README.md:
```diff
@@ -0,0 +1,8 @@
+？？
+# 実行結果
+![image](https://user-images.githubusercontent.com/114993632/199215488-6a907e61-731c-4290-aefc-3c846103db35.png)
+
```
ターミナルを使ってcurlでアクセスするか、またはPOSTMANでアクセスしてみてほしいです。

```bash
curl http://localhost:8080/time
```

toritoripi replied:
![image](https://user-images.githubusercontent.com/114993632/199217380-ddf6fe6c-5565-4c36-b2dd-b8cd8c86a4d4.png)

yoshi-koyama commented on HelloWorldController.java:
```diff
@@ -3,6 +3,11 @@
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RestController;
```
`HelloWorldController.java`が2つありますね。
`package com.example.FirstSBHelloWorld`下のものだけでよいです。

yoshi-koyama replied:
こちらの削除は対応漏れでしょうか？

yoshi-koyama commented on src/main/java/com/example/FirstSBHelloWorld/HelloWorldController.java:
```diff
@@ -3,6 +3,11 @@
 import org.springframework.web.bind.annotation.GetMapping;
```
`package com.example.FirstSBHelloWorld;`
Javaのパッケージ名の命名規則について調べた上で修正お願いします！
参考：https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

yoshi-koyama commented on HelloWorldController.java:
```diff
@@ -3,6 +3,11 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.time.LocalDateTime;
+
```
<img width="280" alt="スクリーンショット 2022-11-01 19 48 34" src="https://user-images.githubusercontent.com/62045457/199216833-8ebc8924-21ef-4c30-8fdc-ebac76898780.png">

いまこれは`main` -> `FirstSB-helloworld`にマージするようになっていますね。
今この状態を直すのは面倒なので今回はこのままでOKです。

次にリポジトリを作るときには`開発ブランチ` -> `main`のようにPull Requestを作りましょう。

toritoripi commented:
皆さんが行っているように、このプルリクエストの中に修正したものをプッシュ？することができません、、、
ターミナルとGitHubを永遠に行ったり来たりしています。もう少し時間がかかりそうです。

toritoripi committed:
HelloWorldController.java を更新

①パッケージ名の変更(アルファベット小文字+アンダースコア)②余分な半角スペースを修正③snake_caseにもとづいて小文字表記(Greetinh→greeting)

toritoripi committed:
Delete HelloWorldController.java

重複していたため削除

toritoripi committed:
Merge branch 'FirstSB-helloworld' into main
